avm/res/compute/gallery/image/main.bicep

```diff
@@ -12,108 +12,13 @@ param location string = resourceGroup().location
-param maxRecommendedvCPUs int = 4
-
-@sys.description('Optional. The minimum amount of RAM in GB recommended for this image.')
-@minValue(1)
-@maxValue(4000)
-param minRecommendedMemory int = 4
-
-@sys.description('Optional. The maximum amount of RAM in GB recommended for this image.')
-@minValue(1)
-@maxValue(4000)
-param maxRecommendedMemory int = 16
-
-@sys.description('''Optional. The hypervisor generation of the Virtual Machine.
-- If this value is not specified, then it is determined by the securityType parameter.
-- If the securityType parameter is specified, then the value of hyperVGeneration will be V2, else V1.
-''')
-@allowed([
-  'V1'
-  'V2'
-])
-param hyperVGeneration string?
-
-@sys.description('Optional. The security type of the image. Requires a hyperVGeneration V2.')
-@allowed([
-  'Standard'
-  'TrustedLaunch'
-  'ConfidentialVM'
-  'ConfidentialVMSupported'
-])
-param securityType string = 'Standard'
-
-@sys.description('Optional. Specifiy if the image supports hibernation.')
-param isHibernateSupported bool = false
-
-@sys.description('''Optional. Specify if the image supports accelerated networking.
-Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, greatly improving its networking performance.
-This high-performance path bypasses the host from the data path, which reduces latency, jitter, and CPU utilization for the most demanding network workloads on supported VM types.
-''')
-param isAcceleratedNetworkSupported bool = true
-
-@sys.description('Optional. The description of this gallery Image Definition resource. This property is updatable.')
-param description string?
-
-@sys.description('Optional. The Eula agreement for the gallery Image Definition. Has to be a valid URL.')
-param eula string?
-
-@sys.description('Optional. The privacy statement uri. Has to be a valid URL.')
-param privacyStatementUri string?
-
-@sys.description('Optional. The release note uri. Has to be a valid URL.')
-param releaseNoteUri string?
-
-@sys.description('Optional. The product ID.')
-param productName string?
-
-@sys.description('Optional. The plan ID.')
-param planName string?
-
-@sys.description('Optional. The publisher ID.')
-param planPublisherName string?
-
-@sys.description('Optional. The end of life date of the gallery Image Definition. This property can be used for decommissioning purposes. This property is updatable. Allowed format: 2020-01-10T23:00:00.000Z.')
-param endOfLife string = ''
-
-@sys.description('Optional. List of the excluded disk types (e.g., Standard_LRS).')
-param excludedDiskTypes array = []
+@description('Required. Properties used to create the image.')
+param galleryImage imageType
 
 @sys.description('Optional. Array of role assignments to create.')
 param roleAssignments roleAssignmentType
 
-@sys.description('Optional. Tags for all resources.')
+@sys.description('Optional. Tags for all resources. This takes precedence over the tags of the galleryImage.tags property.')
```

Comment:
Please note that given that each module is treated like an indipendent unit, the added sentense does not make much sense. It would only if implemented in a UDT of the parent